### ISSUE TITLE
Crono Job deletes all entries if it can't connect to the registry

### ISSUE THREAD

**Issue by Vad1mo** [SUSE/Portus#1292]
I applied the patch from #1184 to our running instance a few weeks ago it looked like it works until two days ago. Now all registry entries (in the DB) are gone! the reason is this:

```
  Crono::CronoJob Load (0.4ms)  SELECT  `crono_jobs`.* FROM `crono_jobs` WHERE `crono_jobs`.`id` = 1 LIMIT 1
I, [2017-06-05T07:25:35.451968 #7]  INFO -- : Perform CatalogJob
  Registry Load (0.4ms)  SELECT  `registries`.* FROM `registries`  ORDER BY `registries`.`id` ASC LIMIT 1000
Exception: Something went wrong while fetching the catalog Response: [401] - {"errors":[{"code":"UNAUTHORIZED","message":"authentication required","detail":[{"Type":"registry","Class":"","Name":"catalog","Action":"*"}]}]}
```

The reason for the 401 is because the docker registry is using old expired certificates.  Crono job interprets this as a zero result for the registry and wipes all entries.  After restart of registry and correct certificates everything is working fine and crono adds all the entries back.

I think that on connection errors to the registry Crono should not treat it as an empty response and wipe the content.

**Comment by mssola** [comment]
I was only able to reproduce this when a 401 was returned when fetching the tags of a repository in the `catalog` call. PR #1293 should be able to fix it. Could you test it whenever you have the time 😋 ?

**Comment by Vad1mo** [comment]
I will give #1293 a try.  Thanks

**Comment by odg0318** [comment]
The PR seems to still work wrong.
If `add_tags` returns nil, expired repositories go to r["tags"].blank? condition in CatalogJob so that they are stuck into dangling list and deleted.
I think that another condition is required when request to registry is expired.

**Comment by mssola** [comment]
This has been greatly alleviated thanks to #1675, which is included in the 2.3 release. For the other corner cases we have #1599, so let's focus on that issue instead.

Thanks all of you for your feedback!
